### ISSUE TITLE
[Bug] Problem accessing server/NAS remote path since installing Kodi v 19.5 to 20.2

### ISSUE THREAD

**Issue by Skyline-007** [xbmc/xbmc#24155]
**## Bug report**
**### Describe the bug**
Good morning

Since installing Kodi version 19.5 to version 20.2

I have a problem with adding sources and my remote paths from my server.

All my sources are on the same server.

I no longer have access to some of my remote paths (my sources) from my server.

Here are my remote paths from my server

See the picture:
![Sans titre2](https://github.com/xbmc/xbmc/assets/108820368/d3ea38b5-f166-4b23-b027-d3578e7c3c7d)

I don't have access to these remote paths:

ftp : // My server address : 21 / 1 Videos summer vacation
ftp : // My server address : 21 / 2 Videos winter holidays

Here is the error message:
![Sans titre1](https://github.com/xbmc/xbmc/assets/108820368/970538c7-6fa9-42ed-8584-00c530b91afe)

But I do have access to the last remote path added last from my server:
ftp : // My server address : 21 /3 Video

![Sans titre](https://github.com/xbmc/xbmc/assets/108820368/1f553529-ccc8-4440-a594-432dd043198e)

There must be an error somewhere since the Kodi v19.5 Update ?

Because before I had access to all my remote paths from my server on old versions of Kodi up to version 19.4 .

Thank you in advance for your answers.

Thanks again for all the work you have done

I reinstalled the old Kodi Version 19.4 and I have no problems.
I have good access to all my remote files, without any error messages.

I hope someone will look at my Debug log v20.2 and Debug log v19.4 to see where the problem is coming from.

This is indeed a bug

I report a bug which has been present since the version of Kodi 19.5 to v20.2

This is not a forum support request, please do not close the post.

**### To Reproduce**
1. Add multiple sources (FTP) Video as written for the problem.

**### Debuglog**
The debuglog can be found here:

[Debug log v20.2](https://paste.kodi.tv/etejocabuf.kodi)

[Debug log v19.4](https://paste.kodi.tv/aliqajajig.kodi)
(kodi v 19.4 I have access to all my remote files without any error messages.)

**### Your Environment**
**Used Operating system:**
 - [x] Windows (Kodi 20.2)
 - [x] Android TV (Kodi 20.2)
 - [x] Raspberry Pi (LibreELEC)

 - Operating system version/name: Kodi 20.2 and Raspberry Pi libreELEC
 - Kodi version: 20.2

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by razzeee** [comment]
Pinging everyone you can think of is NOT okay

**Comment by neo1973** [comment]
> Pinging everyone you can think of is NOT okay

Indeed 🙁 

Regarding the issue: Possibly caused by #21475, there are also comments regarding issues that occurred during testing that could be related.

**Comment by neo1973** [comment]
@Skyline-007: Can you please check if you see incorrect paths or credentials in `${kodi_dir}/userdata/sources.xml` and `${kodi_dir}/userdata/passwords.xml`?

**Comment by Skyline-007** [comment]
> Pinging everyone you can think of is NOT okay

I'm really sorry, but I didn't know how to get my post to be seen by members working on the project.
Because moderation would have closed my post and directed me to Forum Support, even though this is a Bug.



> @Skyline-007: Can you please check if you see incorrect paths or credentials in `${kodi_dir}/userdata/sources.xml` and `${kodi_dir}/userdata/passwords.xml`?

I found the sources.xml file

I looked and all the information is correct.

C:\Users\OneThemAsk\AppData\Roaming\Kodi\userdata\sources.xml

[Sources.xml](https://paste.kodi.tv/urotuluxer)

I found the sources.xml file

I looked and all the information is correct.


However, I couldn't find the passwords.xml file.
Do you know where to find it ? I use Windows 11

There is no password.xml file in C:\Users\OneThemAsk\AppData\Roaming\Kodi\userdata\

**Comment by neo1973** [comment]
> There is no password.xml file in C:\Users\OneThemAsk\AppData\Roaming\Kodi\userdata\

That's fine, it's only created when adding/modifying sources with 19.5 or later.

To narrow this down, can you with Kodi 19.5 or later try changing the passwords of your FTP sources to something different and than change them back? Do this in Kodi, not by modifying the file. As I understand it this should remove the credentials from the sources.xml and create a passwords.xml. Maybe it makes a difference.

**Comment by Skyline-007** [comment]
Kodi 20.2 [password.xml](https://paste.kodi.tv/edacajadoj)

After editing the source passwords.
This shows me a second username (OnTheMask) in the password.xml file but does not show me the third username it replaces the second username with the third username instead of adding lines below for third username (OnTheMask2)
And I only have access to the last source where I changed the password (OnTheMask)

**Comment by Skyline-007** [comment]
I noticed that if I change the password the first username comes in (OneTheMask3).
This shows me the same user below (OneTheMask3)

Except that when I edited the source and put a bad password it displayed it at the top on the first lines.
And once I re-edit the source it shows the same username (OneTheMask3) below with the correct password

So the username (OneTheMask3) and duplicate twice instead of changing the wrong password

Here is the password.xml file Kodi 20.2 [password.xml](https://paste.kodi.tv/vuxezoleqi)



I hope that the different information I give you will help you resolve the problem.

Thanks again for all the work you do.

**Comment by Skyline-007** [comment]
@neo1973 Do you want other information files to help you find what the problem is ? Or is that enough for you for now ?

Thanks again for all the work you do

Best regards

Skyline-007

**Comment by Skyline-007** [comment]
@neo1973 

I have already uninstalled kodi and reinstalled kodi 20.2 and it still gives me the same error message.

I forgot to specify that my sources each have different usernames
Example :
User 1: OnTheMask
User 2: OnTheMask2
User 3: OnTheMask3

There must be a problem with a file that manages the different user assignments ?

**Comment by neo1973** [comment]
I went back to 20.2 and I'm still not able to reproduce the issue. I added similar sources and everything looks like expected:

<details>
<summary>sources.xml</summary>

```xml
<sources>
    <programs>
        <default pathversion="1"></default>
    </programs>
    <video>
        <default pathversion="1"></default>
        <source>
            <name>1 Video summer vacation</name>
            <path pathversion="1">ftp://127.0.0.1:21/1 Video summer vacation/</path>
            <allowsharing>true</allowsharing>
        </source>
        <source>
            <name>2 Videos Winter holidays</name>
            <path pathversion="1">ftp://127.0.0.1:21/2 Videos Winter holidays/</path>
            <allowsharing>true</allowsharing>
        </source>
        <source>
            <name>3 Video</name>
            <path pathversion="1">ftp://127.0.0.1:21/3 Video/</path>
            <allowsharing>true</allowsharing>
        </source>
    </video>
    <music>
        <default pathversion="1"></default>
    </music>
    <pictures>
        <default pathversion="1"></default>
    </pictures>
    <files>
        <default pathversion="1"></default>
    </files>
    <games>
        <default pathversion="1"></default>
    </games>
</sources>
```
</details>

<details>
<summary>passwords.xml</summary>

```xml
<passwords>
    <path>
        <from pathversion="1">ftp://127.0.0.1/1 Video summer vacation</from>
        <to pathversion="1">ftp://OnTheMask:pwd1@127.0.0.1:21/1 Video summer vacation/</to>
    </path>
    <path>
        <from pathversion="1">ftp://127.0.0.1/2 Videos Winter holidays</from>
        <to pathversion="1">ftp://OnTheMask2:pwd2@127.0.0.1:21/2 Videos Winter holidays/</to>
    </path>
    <path>
        <from pathversion="1">ftp://127.0.0.1/3 Video</from>
        <to pathversion="1">ftp://OnTheMask3:pwd3@127.0.0.1:21/3 Video/</to>
    </path>
</passwords>
```
</details>

**Comment by Skyline-007** [comment]
I had to format my computer and I reinstalled Windows 11

I reinstalled Kodi 20.2

I added my 3 sources from my server.
Here are the “sources.xml” and “passwords.xml” files

[sources.xml](https://paste.kodi.tv/owolefohor)

[passwords.xml](https://paste.kodi.tv/uneniyuhoz)


In the "passwords.xml" file we can clearly see that I am missing the following sources: OnTheMask and
OnTheMask2 (missing lines)


When adding sources it records only the last source added last in the "passwords.xml" file.

The "passwords.xml" file removes all other sources added before.

The “passwords.xml” file only remembers the last source (OnTheMask3) added last and deletes the previous sources added….
